embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/arch/DataSource.kt

```diff
@@ -16,3 +30,24 @@ internal interface DataSource {
      */
     fun unregisterListeners()
 }
+
+/**
+ * A [DataSource] that adds either a [EmbraceSpanEvent] or [EmbraceSpanAttribute]
+ * to the current session span.
+ */
+internal interface EventDataSource : DataSource<CurrentSessionSpan>
+
+/**
+ * A [DataSource] that adds or alters a new span on the [SpansService]
+ */
+internal interface SpanDataSource : DataSource<EmbraceTracer>
+
+/**
+ * A [DataSource] that adds a new log to the log service.
+ */
+internal interface LogDataSource : DataSource<LogService>
+
+/**
+ * Placeholder type for the log service.
+ */
+internal interface LogService
```

Comment:
This is just a placeholder type for now, as we haven't defined the contract for what log capture should look like.